### PR TITLE
Refactor Code and add test cases

### PR DESCRIPTION
**Pull request by luii**
Dont merge the PR yet!

### CODE REVIEW

**Comment by luii** [comment]
My Git fucked up sry

File: index.js [hunk 1]
```diff
@@ -84,7 +83,7 @@ function getPercentageDependencies(dependencies, callback)
   {
     if(typeof dependency === 'string') return callback(null, 1)
 
-    calcPercentage(dependency, callback)
+    return calcPercentage(dependency, callback)
```
Comment (piranna) [inline index.js]: Same here

File: index.js [hunk 2]
```diff
@@ -128,17 +127,28 @@ function calcPercentage(module, callback)
...
+ *                               npm-remote-ls package or returning
+ *                               the percentage of the module
+ */
 function coverdeeps(moduleName, callback)
 {
   RemoteLS({development: false}).ls(moduleName, 'latest', function(tree) {
     if(!tree) return callback(new Error(moduleName+' not found on NPM registry'))
 
-    calcPercentage(tree, callback)
+    return calcPercentage(tree, callback)
```
Comment (piranna) [inline index.js]: Same here

File: index.js [hunk 3]
```diff
@@ -128,17 +127,28 @@ function calcPercentage(module, callback)
     module.covered          = result.module
     module.covered_combined = result.module * result.dependencies
 
-    callback(null, module)
+    return callback(null, module)
```
Comment (piranna) [inline index.js]: Same here

File: index.js [hunk 4]
```diff
@@ -53,7 +52,7 @@ function getPercentageProject(repo, callback)
       body = JSON.parse(body)
       if(!body) return callback(null, 0)
 
-      callback(null, body.covered_percent/100)  // Per-unit to easier calcs
+      return callback(null, body.covered_percent/100)  // Per-unit to easier calcs
```
Comment (piranna) [inline index.js]: Why the return? It's not needed...

File: index.js [hunk 5]
```diff
@@ -95,7 +94,7 @@ function getPercentageDependencies(dependencies, callback)
       return prev + (results[key].covered_combined || 0)
     }, 0)
 
-    callback(null, covered / keys.length)
+    return callback(null, covered / keys.length)
```
Comment (piranna) [inline index.js]: Same here

File: index.js (hunk 4 above)
Comment (luii) [inline index.js]: Well i discarded the branch, nevermind

Comment (piranna) [inline index.js]: :+1: So can I close this pull-request?

**Pull request closed without merging**